seatunnel-connectors-v2/connector-google-firestore/src/main/java/org/apache/seatunnel/connectors/seatunnel/google/firestore/sink/FirestoreSinkWriter.java

```diff
@@ -0,0 +1,85 @@
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.google.firestore.sink;
+
+import org.apache.seatunnel.api.table.type.SeaTunnelRow;
+import org.apache.seatunnel.api.table.type.SeaTunnelRowType;
+import org.apache.seatunnel.connectors.seatunnel.common.sink.AbstractSinkWriter;
+import org.apache.seatunnel.connectors.seatunnel.google.firestore.config.FirestoreParameters;
+import org.apache.seatunnel.connectors.seatunnel.google.firestore.exception.FirestoreConnectorErrorCode;
+import org.apache.seatunnel.connectors.seatunnel.google.firestore.exception.FirestoreConnectorException;
+import org.apache.seatunnel.connectors.seatunnel.google.firestore.serialize.DefaultSeaTunnelRowSerializer;
+import org.apache.seatunnel.connectors.seatunnel.google.firestore.serialize.SeaTunnelRowSerializer;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.firestore.CollectionReference;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.FirestoreOptions;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.Base64;
+
+@Slf4j
+public class FirestoreSinkWriter extends AbstractSinkWriter<SeaTunnelRow, Void> {
+
+    private Firestore firestore;
+
+    private CollectionReference collectionReference;
+
+    private SeaTunnelRowSerializer serializer;
+
+    public FirestoreSinkWriter(SeaTunnelRowType seaTunnelRowType, FirestoreParameters parameters)
+            throws IOException {
+        GoogleCredentials credentials;
+        if (parameters.getCredentials() != null) {
+            byte[] bytes = Base64.getDecoder().decode(parameters.getCredentials());
+            credentials = GoogleCredentials.fromStream(new ByteArrayInputStream(bytes));
+        } else {
+            credentials = GoogleCredentials.getApplicationDefault();
+        }
+        FirestoreOptions firestoreOptions =
+                FirestoreOptions.getDefaultInstance()
+                        .toBuilder()
+                        .setProjectId(parameters.getProjectId())
+                        .setCredentials(credentials)
+                        .build();
+        this.firestore = firestoreOptions.getService();
+        this.collectionReference = firestore.collection(parameters.getCollection());
+        this.serializer = new DefaultSeaTunnelRowSerializer(seaTunnelRowType);
+    }
+
+    @Override
+    public void write(SeaTunnelRow seaTunnelRow) throws IOException {
+        collectionReference.add(serializer.serialize(seaTunnelRow));
+    }
+
+    @Override
+    public void close() throws IOException {
+        if (firestore != null) {
+            try {
+                firestore.close();
+            } catch (Exception e) {
+                log.error("Close Firestore client failed.", e);
+                throw new FirestoreConnectorException(
+                        FirestoreConnectorErrorCode.CLOSE_CLIENT_FAILED, e);
```

Comment:
```suggestion
                throw new FirestoreConnectorException(
                        FirestoreConnectorErrorCode.CLOSE_CLIENT_FAILED, "Close Firestore client failed.", e);
```
Logging and throwing exception can cause log duplication.